internal/pkg/networking/resolve.go

```diff
@@ -21,16 +21,26 @@ import (
 	"istio.io/istio/pkg/slices"
 )
 
-func Resolve(addr string) []string {
-	if ip := net.ParseIP(addr); ip != nil {
-		return []string{addr}
-	}
+func Resolve(addrs ...string) []string {
+	var allIPs []string
+	for _, addr := range addrs {
+		if IsIP(addr) {
+			allIPs = append(allIPs, addr)
+			continue
+		}
 
-	ips, err := net.LookupIP(addr)
-	if err != nil {
-		fmt.Printf("Failed to resolve '%s': %v\n", addr, err)
+		ips, err := net.LookupIP(addr)
+		if err != nil {
+			fmt.Printf("Failed to resolve '%s': %v\n", addr, err)
```

Comment:
I think we shouldn't mix logging and printf. Not for this PR per se, but something to unify moving forward.

Reply:
I did it unintentionally. I changed `fmt.Printf` to `log.Errorf` for now, but we will need better solution anyway. I will make another PR with more optimal solution to handling DNS changes.